client/src/features/session/components/SimpleSessionButton.tsx

```diff
@@ -73,7 +79,14 @@ export default function SimpleSessionButton({
 
   if (!runningSession) {
     return (
-      <Link className={className} to={sessionAutostartUrl}>
+      <Link
+        className={className}
+        to={{
+          pathname: sessionNewUrl,
+          search: `autostart=1${fromLanding ? "&fromLanding=1" : ""}`,
```

Comment:
Similar to CSS class names, avoid construction by hand:

```suggestion
          search: new URLSearchParams({
            autostart: "1",
            ...(fromLanding ? { fromLanding: "1" } : {}),
          }).toString(),
```